lib/ddtrace/contrib/rack/request_queue.rb

```diff
@@ -0,0 +1,34 @@
+module Datadog
+  module Contrib
+    module Rack
+      # QueueTime simply...
+      module QueueTime
+        REQUEST_START = 'HTTP_X_REQUEST_START'.freeze
+        QUEUE_START = 'HTTP_X_QUEUE_START'.freeze
+
+        module_function
+
+        def get_request_start(env, now = Time.now.utc)
```

Comment:
Consider moving this to the core library in a future pull request. The behavior here is generic enough, and useful for other integrations.